src/GoTrueClient.ts

```diff
@@ -250,6 +250,34 @@ export default class GoTrueClient {
       return { data: null, user: null, error }
     }
   }
+  /**
+   * Sets the session data from refresh_token
+   * @param refresh_token a jwt token
```

Comment:
Doing this with a refresh token feels a bit funky to me. I'd expect to be able to do this with an access token I think.

Reply:
Agreed - this should be passed a JWT access token (which will be saved inside the cookie). 

Inside a node environment the user will 

- extract the JWT access token from the cookie
- call `setSession`

Thanks for spotting this @dbredvick

Reply:
published a [branch with an alternate implementation](https://github.com/easy-street/gotrue-js/commit/d0806f6031fac882e4cb688ee85ac32fd6e33275), with the steps described here by @kiwicopple as guidance; https://github.com/supabase/gotrue-js/issues/34#issuecomment-823747025_ 

@Iamshankhadeep maybe some inspiration there if you'd like to continue here with the feedback in the comments? or I can create a new PR, lmk.

Reply:
this is looking very good @easy-street 👍

Reply:
@easy-street @kiwicopple I will complete the PR. thanks for all the points.

Reply:
@kiwicopple is there any api endpoint that fetches the `session` over `access_token`? cause looking at @easy-street implementation I see that on session `expires_in` and `expires_at` ans `refresh_token` is null, so the client has no idea that when to call `_callRefreshToken()` to fetch the new `access_token`.

Reply:
Yes you're absolutely right - the refresh token won't be available on the JWT.

OK so it looks like we should change  `setSession()` to `setUser()`, and it will just fill the client with the user credentials (which should then be passed in each HTTP request). 

There will be no way to refresh the session without the access token - but this is fine because it's supposed to be used in a Node context and in that case it will be up to the developer to manage the session/refreshes etc (which they can do with the [api](https://github.com/supabase/gotrue-js/blob/56b6feb44f4e5eb5b166d9512d9a3356bb9e0f0d/src/GoTrueApi.ts#L249)

Reply:
@kiwicopple  So what I suggest is that let's have 2 public function `setSession()` depends on `refresh_token`  and `setUser()` depends on `access_token`. but if we use `setUser()`, `expires_in` and `expires_at` and`refresh_token` will be null on the session data and developer needs to make sure that they call the `api` to manage `sessions`.

Reply:
That sounds solid - pretty sure that will work for #34

Reply:
@kiwicopple can you review the PR?